lib/index.ts

```diff
@@ -0,0 +1,572 @@
+	};
+
+	private apiKey: string;
+
+	constructor(apiKey: string) {
+		// Key.
+		this.apiKey = apiKey;
+
+		// Setup resource paths.
+		this.comment = {
+			create: _.partial(this.httpCall,
+				{ reqType: 'POST', path: 'conversations/{conversation_id}/comments' }),
+			get: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'comments/{comment_id}' }),
+			listMentions: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'comments/{comment_id}/mentions' }),
+		};
+		this.conversation = {
+			get: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'conversations/{conversation_id}' }),
+			list: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'conversations[q{q}][page={page}][limit={limit}]' }),
+			listComments: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'conversations/{conversation_id}/comments' }),
+			listFollowers: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'conversations/{conversation_id}/followers' }),
+			listInboxes: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'conversations/{conversation_id}/inboxes' }),
+			listMessages: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'conversations/{conversation_id}/messages[page={page}][limit={limit}]' }),
+			update: _.partial(this.httpCall,
+				{ reqType: 'PATCH', path: 'conversations/{conversation_id}' }),
+		};
+		this.inbox = {
+			create: _.partial(this.httpCall,
+				{ reqType: 'POST', path: 'inboxes' }),
+			get: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'inboxes/{inbox_id}' }),
+			list: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'inboxes' }, null),
+			listChannels: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'inboxes/{inbox_id}/channels' }),
+			listConversations: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'inboxes/{inbox_id}/conversations[q{q}...][page={page}][limit={limit}]' }),
+			listTeammates: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'inboxes/{inbox_id}/teammates' }),
+		};
+		this.message = {
+			get: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'messages/{message_id}' }),
+			receiveCustom: _.partial(this.httpCall,
+				{ reqType: 'POST', path: 'channels/{channel_id}/incoming_messages' }),
+			reply: _.partial(this.httpCall,
+				{ reqType: 'POST', path: 'conversations/{conversation_id}/messages' }),
+			send: _.partial(this.httpCall,
+				{ reqType: 'POST', path: 'channels/{channel_id}/messages' }),
+		};
+	}
+
+	private httpCall = (details: FrontRequest, params: any, callback?: InternalCallback): Promise<any | void> => {
+		const requestOpts = {
+			body: (params) ? params : {},
+			headers: {
+				Authorization: `Bearer ${this.apiKey}`
+			},
+			json: true,
+			method: details.reqType,
+			url: `${FrontURL}/${this.formatPath(details.path, params)}`
+		};
+
+		// Make the request.
+		return request(requestOpts).promise().catch((error: any) => {
+			// Format this into something useful.
+			throw error.error._error;
```

Comment:
I think I misread my own code here (https://github.com/resin-io/resin-integrations/blob/master/lib/clients/front.js#L30-L36) and interpreted it as a Front inconsistency.

Regarding my second point, I have no idea what the `error` argument in `catch` is. It seems `request-promise` wraps it in some custom object? Hence my question about digging into `error.error._error`.

Regarding my third point, it seems it still stands: based on the output you pasted, it seems `throw error.error._error` indeed throws an object other than `Error`?